src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -14,45 +14,505 @@ IPhysicalDevice::IPhysicalDevice(core::smart_refctd_ptr<system::ISystem>&& s, co
+    if (m_features.descriptorBindingUpdateUnusedWhilePending) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_DESCRIPTOR_BINDING_UPDATE_UNUSED_WHILE_PENDING");
+    if (m_features.descriptorBindingPartiallyBound) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_DESCRIPTOR_BINDING_PARTIALLY_BOUND");
+    if (m_features.descriptorBindingVariableDescriptorCount) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_DESCRIPTOR_BINDING_VARIABLE_DESCRIPTOR_COUNT");
+    if (m_features.runtimeDescriptorArray) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RUNTIME_DESCRIPTOR_ARRAY");
+    if (m_features.samplerFilterMinmax) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SAMPLER_FILTER_MINMAX");
+    if (m_features.scalarBlockLayout) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SCALAR_BLOCK_LAYOUT");
+    if (m_features.uniformBufferStandardLayout) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_UNIFORM_BUFFER_STANDARD_LAYOUT");
+    if (m_features.shaderSubgroupExtendedTypes) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_SUBGROUP_EXTENDED_TYPES");
+    if (m_features.separateDepthStencilLayouts) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SEPARATE_DEPTH_STENCIL_LAYOUTS");
+    if (m_features.bufferDeviceAddress) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_BUFFER_DEVICE_ADDRESS");
+    if (m_features.bufferDeviceAddressMultiDevice) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_BUFFER_DEVICE_ADDRESS_MULTI_DEVICE");
+    if (m_features.vulkanMemoryModel) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_VULKAN_MEMORY_MODEL");
+    if (m_features.vulkanMemoryModelDeviceScope) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_VULKAN_MEMORY_MODEL_DEVICE_SCOPE");
+    if (m_features.vulkanMemoryModelAvailabilityVisibilityChains) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_VULKAN_MEMORY_MODEL_AVAILABILITY_VISIBILITY_CHAINS");
+    if (m_features.subgroupBroadcastDynamicId) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SUBGROUP_BROADCAST_DYNAMIC_ID");
+    if (m_features.shaderDemoteToHelperInvocation) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_DEMOTE_TO_HELPER_INVOCATION");
+    if (m_features.shaderTerminateInvocation) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_TERMINATE_INVOCATION");
+    if (m_features.subgroupSizeControl) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SUBGROUP_SIZE_CONTROL");
+    if (m_features.computeFullSubgroups) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_COMPUTE_FULL_SUBGROUPS");
+    if (m_features.shaderIntegerDotProduct) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_INTEGER_DOT_PRODUCT");
+    if (m_features.rasterizationOrderColorAttachmentAccess) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RASTERIZATION_ORDER_COLOR_ATTACHMENT_ACCESS");
+    if (m_features.rasterizationOrderDepthAttachmentAccess) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RASTERIZATION_ORDER_DEPTH_ATTACHMENT_ACCESS");
+    if (m_features.rasterizationOrderStencilAttachmentAccess) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RASTERIZATION_ORDER_STENCIL_ATTACHMENT_ACCESS");
```

Comment:
what do these extensions do?

Reply:
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_ARM_rasterization_order_attachment_access.html
https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#renderpass-feedbackloop

this seems to be for defining a render order to avoid data races when using a feedback loop in render passes, its arm only